src/sinks/http.rs

```diff
@@ -279,14 +293,15 @@ impl util::http::HttpSink for HttpSink {
                     // Prepend before building a request body to eliminate the
                     // additional copy here.
                     let message = body.split();
+                    body.put(self.payload_prefix.as_bytes());
                     body.put_u8(b'[');
                     if !message.is_empty() {
                         body.unsplit(message);
                         // remove trailing comma from last record
                         body.truncate(body.len() - 1);
                     }
                     body.put_u8(b']');
-
+                    body.put(self.payload_suffix.as_bytes());
```

Comment:
🤔 I'm thinking maybe we should sanitize the `payload_prefix` and `payload_suffix` to make sure they start with an opening bracket and end with a closing bracket respectively. I think that is a requirement for the request to succeed, is that true?
I don't think we need to validate the entire contents for correctness, that would be too much.

Would probably also make sense to confirm that if one is non-empty string, that the other is also non-empty string, since that would make malformed JSON.

If those cases are detected (I think it should be checked in the sink start-up), it makes sense to me to error out because unless I'm mistaken, all the HTTP requests would fail due to malformed JSON.

If this is the case, it would be best to add more test cases for validating those checks work. As well as making a note of that in the comment for the config options.

Reply:
I added full json validation, seemed not to hard. probably the validation should only happen if json and characterDelimited framing is used, right?

Reply:
added. I'm only validating if it's comma separated, otherwise the json validation wouldn't work anyway.